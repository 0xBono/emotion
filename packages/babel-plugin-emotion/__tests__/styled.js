import babelTester from 'babel-tester'
import plugin from 'babel-plugin-emotion'
import path from 'path'

babelTester(
  '@emotion/babel-plugin-core styled',
  path.join(
    __dirname,
    '..',
    '..',
    '..',
    'next-packages',
    'styled.macro',
    '__tests__'
  ),
  {
    plugins: [plugin],
    transform: src => src.replace(/\.macro/g, '')
  }
)
