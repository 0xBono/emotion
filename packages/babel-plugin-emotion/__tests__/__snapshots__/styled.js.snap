// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`@emotion/babel-plugin-core styled basic 1`] = `
"import styled from '@emotion/styled'

const SomeComponent = styled.div\`
  color: hotpink;
\`


      ↓ ↓ ↓ ↓ ↓ ↓

import _styled from \\"@emotion/styled-base\\";

const SomeComponent = _styled(\\"div\\", {
  target: \\"eddbda30\\",
  label: \\"SomeComponent\\"
})({
  name: \\"1lrxbo5\\",
  styles: \\"color:hotpink;\\"
});"
`;

exports[`@emotion/babel-plugin-core styled call-expression 1`] = `
"import styled from '@emotion/styled'

const SomeComponent = styled('div')\`
  color: hotpink;
\`


      ↓ ↓ ↓ ↓ ↓ ↓

import _styled from \\"@emotion/styled-base\\";

const SomeComponent =
/*#__PURE__*/
_styled('div', {
  target: \\"e1wwfyhh0\\",
  label: \\"SomeComponent\\"
})({
  name: \\"1lrxbo5\\",
  styles: \\"color:hotpink;\\"
});"
`;

exports[`@emotion/babel-plugin-core styled comments 1`] = `
"import styled from '@emotion/styled'

styled.div\`
  // display:flex;

  /*
wef

dfwf */
  color: hotpink;
\`


      ↓ ↓ ↓ ↓ ↓ ↓

import _styled from \\"@emotion/styled-base\\";

_styled(\\"div\\", {
  target: \\"e1ewmxpp0\\"
})({
  name: \\"1lrxbo5\\",
  styles: \\"color:hotpink;\\"
});"
`;

exports[`@emotion/babel-plugin-core styled component-selector 1`] = `
"import styled from '@emotion/styled'

const SomeComponent = styled.div\`
  color: hotpink;
\`

let SomeOtherComponent = styled.div({ color: 'hotpink' })

let AnotherComponent = styled.div\`
  \${SomeComponent} {
    color: green;
  }
  \${SomeOtherComponent} {
    color: green;
  }
\`

let OneLastComponent = styled.div({
  [SomeComponent]: {
    color: 'green'
  },
  [SomeOtherComponent]: {
    color: 'green'
  }
})


      ↓ ↓ ↓ ↓ ↓ ↓

import _styled from \\"@emotion/styled-base\\";

const SomeComponent = _styled(\\"div\\", {
  target: \\"e14x8jfs0\\",
  label: \\"SomeComponent\\"
})({
  name: \\"1lrxbo5\\",
  styles: \\"color:hotpink;\\"
});

let SomeOtherComponent = _styled(\\"div\\", {
  target: \\"e14x8jfs1\\",
  label: \\"SomeOtherComponent\\"
})({
  name: \\"1lrxbo5\\",
  styles: \\"color:hotpink;\\"
});

let AnotherComponent = _styled(\\"div\\", {
  target: \\"e14x8jfs2\\",
  label: \\"AnotherComponent\\"
})(SomeComponent, \\"{color:green;}\\", SomeOtherComponent, \\"{color:green;}\\");

let OneLastComponent = _styled(\\"div\\", {
  target: \\"e14x8jfs3\\",
  label: \\"OneLastComponent\\"
})({
  [SomeComponent]: {
    color: 'green'
  },
  [SomeOtherComponent]: {
    color: 'green'
  }
});"
`;

exports[`@emotion/babel-plugin-core styled existing-options 1`] = `
"import styled from '@emotion/styled'

styled('div', { shouldForwardProp: window.whatever }, window.whatever)()

styled('div', { shouldForwardProp: window.whatever }, window.whatever)\`\`

styled(
  window.whatever,
  { shouldForwardProp: window.whatever },
  window.whatever
)()

styled(
  window.whatever,
  { shouldForwardProp: window.whatever },
  window.whatever
)\`\`


      ↓ ↓ ↓ ↓ ↓ ↓

import _styled from \\"@emotion/styled-base\\";

/*#__PURE__*/
_styled('div', {
  shouldForwardProp: window.whatever,
  target: \\"e1pu1ggq0\\"
}, window.whatever)();

/*#__PURE__*/
_styled('div', {
  shouldForwardProp: window.whatever,
  target: \\"e1pu1ggq1\\"
}, window.whatever)();

/*#__PURE__*/
_styled(window.whatever, {
  shouldForwardProp: window.whatever,
  target: \\"e1pu1ggq2\\"
}, window.whatever)();

/*#__PURE__*/
_styled(window.whatever, {
  shouldForwardProp: window.whatever,
  target: \\"e1pu1ggq3\\"
}, window.whatever)();"
`;

exports[`@emotion/babel-plugin-core styled function-interpolation 1`] = `
"import styled from '@emotion/styled'

let Avatar = styled.img\`
  width: 96px;
  height: 96px;
  border-radius: \${props => props.theme.borderRadius};
  border: 1px solid \${props => props.theme.borderColor};
\`


      ↓ ↓ ↓ ↓ ↓ ↓

import _styled from \\"@emotion/styled-base\\";

let Avatar = _styled(\\"img\\", {
  target: \\"e1iwlc0p0\\",
  label: \\"Avatar\\"
})(\\"width:96px;height:96px;border-radius:\\", props => props.theme.borderRadius, \\";border:1px solid \\", props => props.theme.borderColor, \\";\\");"
`;

exports[`@emotion/babel-plugin-core styled label 1`] = `
"import styled from '@emotion/styled'

class Thing {
  static SomeComponent = styled.div\`
    color: hotpink;
  \`
  BadIdeaComponent = styled.div\`
    background: hotpink;
  \`
}


      ↓ ↓ ↓ ↓ ↓ ↓

import _styled from \\"@emotion/styled-base\\";

class Thing {
  static SomeComponent = _styled(\\"div\\", {
    target: \\"erkd9g00\\",
    label: \\"SomeComponent\\"
  })({
    name: \\"1lrxbo5\\",
    styles: \\"color:hotpink;\\"
  });
  BadIdeaComponent = _styled(\\"div\\", {
    target: \\"erkd9g01\\",
    label: \\"BadIdeaComponent\\"
  })({
    name: \\"d1m7sn\\",
    styles: \\"background:hotpink;\\"
  });
}"
`;

exports[`@emotion/babel-plugin-core styled more-than-10-interpolations 1`] = `
"import styled from '@emotion/styled'

// yes, this was actually a bug at one point
const H1 = styled.h1\`
  text-decoration: \${'underline'};
  border-right: solid blue \${54}px;
  background: \${'white'};
  color: \${'black'};
  display: \${'block'};
  border-radius: \${'3px'};
  padding: \${'25px'};
  width: \${'500px'};
  z-index: \${100};
  font-size: \${'18px'};
  text-align: \${'center'};
  border-left: \${p => p.theme.blue};
\`


      ↓ ↓ ↓ ↓ ↓ ↓

import _styled from \\"@emotion/styled-base\\";

// yes, this was actually a bug at one point
const H1 = _styled(\\"h1\\", {
  target: \\"e1ip2xek0\\",
  label: \\"H1\\"
})(\\"text-decoration:underline;border-right:solid blue \\", 54, \\"px;background:white;color:black;display:block;border-radius:3px;padding:25px;width:500px;z-index:\\", 100, \\";font-size:18px;text-align:center;border-left:\\", p => p.theme.blue, \\";\\");"
`;

exports[`@emotion/babel-plugin-core styled multiple-calls 1`] = `
"import styled from '@emotion/styled'

const SomeComponent = styled.div\`
  color: hotpink;
\`

const SomeOtherComponent = styled.button\`
  color: green;
\`


      ↓ ↓ ↓ ↓ ↓ ↓

import _styled from \\"@emotion/styled-base\\";

const SomeComponent = _styled(\\"div\\", {
  target: \\"e15zzbw00\\",
  label: \\"SomeComponent\\"
})({
  name: \\"1lrxbo5\\",
  styles: \\"color:hotpink;\\"
});

const SomeOtherComponent = _styled(\\"button\\", {
  target: \\"e15zzbw01\\",
  label: \\"SomeOtherComponent\\"
})({
  name: \\"bjcoli\\",
  styles: \\"color:green;\\"
});"
`;

exports[`@emotion/babel-plugin-core styled object 1`] = `
"import styled from '@emotion/styled'

const SomeComponent = styled.div({ color: 'hotpink' })


      ↓ ↓ ↓ ↓ ↓ ↓

import _styled from \\"@emotion/styled-base\\";

const SomeComponent = _styled(\\"div\\", {
  target: \\"es0jczh0\\",
  label: \\"SomeComponent\\"
})({
  name: \\"1lrxbo5\\",
  styles: \\"color:hotpink;\\"
});"
`;

exports[`@emotion/babel-plugin-core styled object-call-expression 1`] = `
"import styled from '@emotion/styled'

const SomeComponent = styled('div')({ color: 'hotpink' })


      ↓ ↓ ↓ ↓ ↓ ↓

import _styled from \\"@emotion/styled-base\\";

const SomeComponent =
/*#__PURE__*/
_styled('div', {
  target: \\"e1q97y6o0\\",
  label: \\"SomeComponent\\"
})({
  name: \\"1lrxbo5\\",
  styles: \\"color:hotpink;\\"
});"
`;

exports[`@emotion/babel-plugin-core styled object-function 1`] = `
"import styled from '@emotion/styled'

const SomeComponent = styled.div(props => ({ color: props.color }))


      ↓ ↓ ↓ ↓ ↓ ↓

import _styled from \\"@emotion/styled-base\\";

const SomeComponent = _styled(\\"div\\", {
  target: \\"e18gi6040\\",
  label: \\"SomeComponent\\"
})(props => ({
  color: props.color
}));"
`;

exports[`@emotion/babel-plugin-core styled shorthand-property 1`] = `
"import styled from '@emotion/styled'

let fontSize = window.whatever

const H1 = styled.h1({ fontSize })


      ↓ ↓ ↓ ↓ ↓ ↓

import _styled from \\"@emotion/styled-base\\";
let fontSize = window.whatever;

const H1 = _styled(\\"h1\\", {
  target: \\"e1aegz740\\",
  label: \\"H1\\"
})({
  fontSize
});"
`;

exports[`@emotion/babel-plugin-core styled with-spread 1`] = `
"import styled from '@emotion/styled'

const H1 = styled.h1({ ...window.whatever })


      ↓ ↓ ↓ ↓ ↓ ↓

import _styled from \\"@emotion/styled-base\\";

const H1 = _styled(\\"h1\\", {
  target: \\"e1xgd01a0\\",
  label: \\"H1\\"
})({ ...window.whatever
});"
`;
